File: ios/Modules/Core/Sources/Models/SourceInfo.swift
// Atlas Platform
import Foundation

/// Swift mirror of `atlas_spec::types::source_info::SourceInfo`.
public struct SourceInfo: Codable, Sendable, Identifiable {
    public let id: String
    public var name: String
    public var version: String
    public var lang: String
    public var baseURLs: [String]
    public var contentType: ContentType
    public var supportsNSFW: Bool
    public var capabilities: [String]
    public var iconURL: String?
    public var description: String?

    enum CodingKeys: String, CodingKey {
        case id, name, version, lang
        case baseURLs       = "base_urls"
        case contentType    = "content_type"
        case supportsNSFW   = "supports_nsfw"
        case capabilities
        case iconURL        = "icon_url"
        case description
    }
}

File: crates/atlas-spec/src/types/source_info.rs
use serde::{Deserialize, Serialize};
use super::manga::ContentType;
use crate::capability::Capability;

/// Static metadata about a source, returned by `get_info`.
/// This is the source's identity card.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SourceInfo {
    /// Globally unique identifier, e.g. "mangadex".
    pub id: String,
    /// Display name shown in the UI.
    pub name: String,
    /// SemVer string, e.g. "1.2.0".
    pub version: String,
    /// BCP-47 language tag for the source's primary content language.
    pub lang: String,
    /// Base URLs this source fetches from (used for domain allowlist enforcement).
    pub base_urls: Vec<String>,
    pub content_type: ContentType,
    pub supports_nsfw: bool,
    /// Capabilities this source uses (must match manifest declaration).
    pub capabilities: Vec<Capability>,
    pub icon_url: Option<String>,
    pub description: Option<String>,
}
